### PR TITLE
Add lgc test for CS built-in inputs

### PR DESCRIPTION
**Pull request by trenouf**

### CODE REVIEW

File: lgc/tool/lgc/lgc.cpp [hunk 1]
```diff
@@ -188,7 +190,8 @@ int main(int argc, char **argv) {
       if (notSpacePos != StringRef::npos) {
         if (remaining[notSpacePos] == '!')
           hadMetadata = true;
-        else if (hadMetadata && remaining.slice(notSpacePos, StringRef::npos).startswith("target")) {
+        else if (hadMetadata && (remaining.slice(notSpacePos, StringRef::npos).startswith("target") ||
+                                 remaining.slice(notSpacePos, StringRef::npos).startswith("define"))) {
```
Comment (kuhar) [inline lgc/tool/lgc/lgc.cpp:194]: nit: use StringRef::dropfront?

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6b1ac2638cb17e04b28ba76edded97f03cd8e709
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 16871e90b0f2e6937aecee415af4d2556384bd7f</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev f623fc7ca78c3cec2123a16700a8819f3ccf0656</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 6d7376cb11eeeb48833cd87a055249b7e0a8d9b5</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/144501)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26321/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21853/48174 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26320/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21854/48174 (45.4%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24852/48153 (51.6%)</li>
<li>Failed: 0/48153 (0.0%)</li>
<li>Skipped: 23301/48153 (48.4%)</li>
</ul></details>


</ul></details>

**Comment by trenouf** [comment]
Out of disk space failure. Everything else passed.

**Comment by trenouf** [comment]
V2: Address review comments.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7bb26a41a281053c08c2ccaae20282fa85091f4a
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 16871e90b0f2e6937aecee415af4d2556384bd7f</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev f623fc7ca78c3cec2123a16700a8819f3ccf0656</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 6d7376cb11eeeb48833cd87a055249b7e0a8d9b5</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/144501)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26321/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21853/48174 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26320/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21854/48174 (45.4%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24852/48153 (51.6%)</li>
<li>Failed: 0/48153 (0.0%)</li>
<li>Skipped: 23301/48153 (48.4%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Comment by trenouf** [comment]
Rebase.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4a28350deaec17ff2db6bb00f18c65adcd4e9466
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 6f6b37dd68ffe37868b14bfc1be7cf029eecc361</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev cb0215b2c243fb7919397e9ad0bbce2ac9c7bbc9</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev fde3d4e2bd90743606ca7b724bf62a53cacf4615</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/144501)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26321/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21853/48174 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26320/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21854/48174 (45.4%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24852/48153 (51.6%)</li>
<li>Failed: 0/48153 (0.0%)</li>
<li>Skipped: 23301/48153 (48.4%)</li>
</ul></details>


</ul></details>

**Pull request merged**